torchvision/transforms/functional_tensor.py

```diff
@@ -880,7 +885,7 @@ def perspective(
     )
 
     ow, oh = img.shape[-1], img.shape[-2]
-    grid = _perspective_grid(perspective_coeffs, ow=ow, oh=oh)
+    grid = _perspective_grid(perspective_coeffs, ow=ow, oh=oh, device=img.device)
```

Comment:
For a future PR: I'm not sure this will work as expected if the input tensor is float64 (or float16).